instrumentation/weblogic/src/main/java/com/splunk/opentelemetry/instrumentation/weblogic/metrics/ThreadPoolMetrics.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.weblogic.metrics;
+
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.ThreadPoolSemanticConventions.EXECUTOR_NAME;
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.ThreadPoolSemanticConventions.EXECUTOR_TYPE;
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.ThreadPoolSemanticConventions.TASKS_COMPLETED;
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.ThreadPoolSemanticConventions.THREADS_CURRENT;
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.ThreadPoolSemanticConventions.THREADS_IDLE;
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.jmx.JmxAttributesHelper.getNumberAttribute;
+import static java.util.Arrays.asList;
+
+import com.splunk.opentelemetry.javaagent.bootstrap.jmx.JmxQuery;
+import com.splunk.opentelemetry.javaagent.bootstrap.metrics.jmx.JmxMetricsWatcher;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+public final class ThreadPoolMetrics {
+
+  private static final AtomicBoolean initialized = new AtomicBoolean();
+  private static final JmxMetricsWatcher jmxMetricsWatcher =
+      new JmxMetricsWatcher(
+          JmxQuery.create("com.bea", "Name", "ThreadPoolRuntime"), ThreadPoolMetrics::createMeters);
+
+  public static void initialize() {
+    if (initialized.compareAndSet(false, true)) {
+      jmxMetricsWatcher.start();
+    }
+  }
+
+  private static List<Meter> createMeters(MBeanServer mBeanServer, ObjectName objectName) {
+    Tags tags = executorTags(objectName);
+    return asList(
+        THREADS_CURRENT.create(
+            tags, mBeanServer, getNumberAttribute(objectName, "ExecuteThreadTotalCount")),
+        TASKS_COMPLETED.create(
+            tags,
+            () ->
```

Comment:
How about adding a
```java
  public <T> FunctionCounter create(Tags additionalTags, T obj, ToDoubleFunction<T> function)
```
function to `CounterSemanticConvention`? Similar to what `GaugeSemanticConvention` has